woocommerce/woocommerce-android
# Issue/3725 check location permissions

Parent issue #3725 

Merge instructions:
1. ~Make sure https://github.com/woocommerce/woocommerce-android/pull/3970 is merged~
2. ~Remove "do not merge" label~
3. Merge this PR

This PR adds check for location permissions when the user enters "connect reader" flow. The permissions are required by StripeTerminalSDK.

Logic summary
- user has the permissions, start the scan
- user doesn't have the permissions, request them
- user grants the permissions, start the scan
- user doesn't grant the permissions or they check "never ask again", show missing permissions screen

To test
- clear app data or disable "location" permissions
1. Open app settings
2. Tap on "Manage card reader
3. Tap on "Redirect to CardReaderDetailFragment"
4. Tap on "Connect"
5. Notice the "Request permissions" dialog is shown
6. Refuse to give the app the permissions
7. Notice "Missing permissions" error shown
8. Tap on "Open App settings"
9. Notice you are navigated to system settings -> give the app the permissions

- Repeat steps 1 - 5 but give the apps the permissions in step 6



Update release notes:
